tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -202,3 +218,404 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
       assert.include hls, anchor12.highlight[anchor12.startPage]
       assert.include hls, anchor2.highlight[anchor2.startPage]
       assert.notInclude hls, anchor3.highlight[anchor3.startPage]
+
+  describe 'two-phased anchoring', ->
+
+    class DummyDocumentAccess
+
+      @applicable: -> true
+
+      isPageMapped: (index) -> index in @_rendered
+
+      constructor: ->
+        @_rendered = []
+
+    # Helper function to trigger a page rendering
+    renderPage = (doc, index) ->
+      return if doc.isPageMapped index
+
+      doc._rendered.push index
+
+      # Publish an event
+      event = document.createEvent "UIEvents"
+      event.initUIEvent "docPageMapped", false, false, window, 0
+      event.pageIndex = index
+      window.dispatchEvent event
+
+    # Helper function to trigger a page unrendering
+    unrenderPage = (doc, index) ->
+      return unless doc.isPageMapped index
```

Comment:
> Let's get rid of these statements,

You mean the checks against calling these helper functions in the incorrect state?

> they're unnecessary if we set up the test correctly.

Yes, they are not necessary in the strict sense, but if remove them, and someone write incorrect tests, he can trigger weird things that never happen normally. (Like, rendering of a page that is already rendered.) Shouldn't we have some protection against that?